src/main/scala/basics/Collections.scala

```diff
@@ -0,0 +1,100 @@
+package basics
+
+import scala.annotation.tailrec
+
+object Collections extends App {
+
+  /*
+     In a sorted list find two numbers which have a gap between
+        None for List(1, 2, 3, 4)
+        Some((2, 8)) for List(1, 2, 8)
+   */
+  @tailrec
+  def findGap(l: List[Int]): Option[(Int, Int)] = l match {
+    case first :: second :: tail =>
+      if (second - first > 1) Option(first, second) else findGap(second :: tail)
```

Comment:
Usually we use `Some` constructor and not `Option`

Reply:
Agree, I replaced this solution with more beautiful